src/redux/features/editProfileSlice.ts

```diff
@@ -0,0 +1,248 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+import { sliceNames } from '~/redux/redux.constants'
+import {
+  SubjectNameInterface,
+  UpdatedPhoto,
+  UserMainSubject,
+  UserMainSubjectFieldValues
+} from '~/types'
+
+interface EditProfileState {
+  firstName: string
+  lastName: string
+  country: string
+  city: string
+  professionalSummary: string
+  nativeLanguage: string
+  videoLink: string
+  photo: string | UpdatedPhoto | null
+  categories: UserMainSubject[]
```

Comment:
I think it's better to rename this field to UserMainSubjects or something similar because this field stores not only  categories

Reply:
I named this field "categories" because it is the most concise name which also consistent with UI and other parts of code
### ProfessionalInfoTab (part of EditProfile component):
![Screenshot 2024-07-04 at 12 38 46](https://github.com/ita-social-projects/SpaceToStudy-Client/assets/74725159/ede37c46-fae6-4f1d-847c-d7d8fa32bf6f)
### UI:
![Screenshot 2024-07-04 at 12 38 02](https://github.com/ita-social-projects/SpaceToStudy-Client/assets/74725159/1bc08c2e-e150-496c-987f-a0077fc198f8)